mcstatus/motd.py

```diff
@@ -0,0 +1,820 @@
+
+        try:
+            rgb = tuple(int(hex[i : i + 2], 16) for i in (0, 2, 4))
+        except ValueError:
+            raise ValueError(f"Failed to parse given hex color: {'#' + hex!r}")
+
+        return cls.from_rgb(rgb)
+
+    @classmethod
+    def from_rgb(cls, rgb: tuple[int, int, int]) -> Self:
+        """Parse RGB color, from hex.
+
+        :raises ValueError: When RGB color is out of its 8-bit range.
+        :returns: New ``WebColor`` instance.
+        """
+        if not all(255 >= e >= 0 for e in rgb):
+            raise ValueError("Got RGB color out of its 8-bit range (0-255)")
+
+        hex = "#{:02x}{:02x}{:02x}".format(*rgb)
+        return cls(hex, rgb)
+
+
+@dataclass
+class TranslationTag:
+    """Represents a ``translate`` field in server's answer.
+
+    This just exist, and completely ignored by our end-parsers (``ansi``, ``html`` etc. properties).
+    You can parse them with ``Motd.parsed`` attribute.
+
+    .. seealso:: `Minecraft's wiki. <https://minecraft.fandom.com/wiki/Raw_JSON_text_format#Translated_Text>`_
+    """
+
+    id: str
+
+
+ParsedMotdItem: TypeAlias = "Formatting | MinecraftColor | WebColor | TranslationTag | str"
+
+
+@dataclass
+class Motd:
+    """Represents parsed MOTD.
+
+    Basing on this class, you can write your own MOTD-to-something parser, see ``parsed`` field.
+
+    :param parsed: Parsed MOTD, which then will be transformed.
+        Bases on this field, you can easily write your own MOTD-to-something parser.
+    :param bedrock: Is server Bedrock Edition? Some details may change in work of this class.
+    """
+
+    parsed: list[ParsedMotdItem]
+    bedrock: bool = False
+
+    # Replace those if you want custom transform logic
+    HTML_TRANSFORMER: type[HtmlTransformer] | None = field(repr=False, hash=False, compare=False, default=None)
+    ANSI_3_BIT_TRANSFORMER: type[Ansi3BitTransformer] | None = field(repr=False, hash=False, compare=False, default=None)
+    ANSI_4_BIT_TRANSFORMER: type[Ansi4BitTransformer] | None = field(repr=False, hash=False, compare=False, default=None)
+    ANSI_8_BIT_TRANSFORMER: type[Ansi8BitTransformer] | None = field(repr=False, hash=False, compare=False, default=None)
+    ANSI_24_BIT_TRANSFORMER: type[Ansi24BitTransformer] | None = field(repr=False, hash=False, compare=False, default=None)
+
+    @classmethod
+    def parse(cls, raw: RawResponseDescription, *, bedrock: bool = False) -> Self:
+        """Parse a raw MOTD to less raw MOTD (``Motd.parsed`` field).
+
+        :param raw: Raw MOTD, directly from server.
+        :param bedrock: Is server Bedrock Edition? Nothing changes here, just sets attribute.
+        :returns: New ``MOTD`` instance.
+        """
+        if isinstance(raw, list):
+            raw = RawResponseDescriptionWhenDict(**{"extra": raw})
+
+        if isinstance(raw, str):
+            parsed = cls._parse_as_str(raw, bedrock=bedrock)
+        else:
+            parsed = cls._parse_as_dict(raw, bedrock=bedrock)
```

Comment:
I think this might be worth keeping around, so that we know about errors where motd contained a color that we weren't able to parse. This might for example happen if some motd would use a different color format (like `#FFF`, instead of what we'd expect, being `#FFFFFF`), or anything like this. The error makes it clear where the mistake occured, and that it's likely something that the library is expected to fix, as it means we're not parsing all of the possible colors the motd can have, I think it should stay.

As for the first example, yeah, that one is not as important, but it's not really hurting anything, and it's just how I prefer to write these functions, I hate using else with isinstance checks, to assume the other option(s) from the type-hint, I prefer to be explicit and raise an exception otherwise. However I don't mind it if you don't include this error and keep this as it was with the first one.

Reply:
Fixed [here](https://github.com/py-mine/mcstatus/compare/c5d0a9f4db39c4962022ae57ab6e295f1faceab0..5761f0d4e4a8c1580071922aa815a849aad2873d).